syntax = "proto2";
package netData;

// 3-1 ping
message Ping
{
	
}

// 3-2 通知Client数据加载完毕
message LoadPlayerDataFinish
{
	enum EC
	{
		EC_OTHER = 0;
		EC_SUCC = 1;
		EC_FAIL = 2;
	};
	optional int32 nCode = 1;	// 返回的信息
	optional uint32 nTempID = 2;
}

// 3-3 请求移动
message PlayerMove
{
	optional float x = 1;
	optional float y = 2;
	optional float z = 3;
}

// 3-4 移动返回
message PlayerMoveRet
{
	optional float x = 1;
	optional float y = 2;
	optional float z = 3;
	optional uint32 nTempID = 4;//对象的临时ID
	optional int32 nCode = 5;//返回的错误码
}

// 3-5 请求切换地图
message ChangeMap
{
	optional int32 nMapID = 1;
	optional string sEtoken = 2;	// 账号信息,前端不用填写
	optional bytes sSecret = 3;	// 秘钥,前端不用填写
}

// 3-6 切换地图返回，如果涉及到换线，那么不会返回该消息，返回的是2-8
message ChangeMapRet
{
	enum EC
	{
		EC_OTHER = 0;
		EC_SUCC = 1;
		EC_FAIL = 2;
		EC_LINE = 4;	//线路不存在
		EC_MAP = 5;		//地图不存在
		EC_LEAVEMAP = 6;//离开地图失败
		EC_PACKDATA = 7;//打包数据失败
	};
	optional int32 nCode = 1;		// 返回的信息
	optional int32 nMapID = 2;		// 要加载的地图ID
	optional int32 nTempID = 3;		// 新的tempid
}

// 3-7 更新对象信息（有则更新，无则添加）
message UpdataObjInfo
{
	optional float x = 1;
	optional float y = 2;
	optional float z = 3;
	optional uint32 nTempID = 4;//对象的临时ID
	optional string Name = 5;//名称
}

// 3-8 删除视野中的对象
message DelObjFromView
{
	optional uint32 nTempID = 1;//对象的临时ID
}